libpromises/crypto.c

```diff
@@ -87,10 +87,25 @@ void CryptoDeInitialize()
 {
     if (crypto_initialized)
     {
+        char randfile[CF_BUFSIZE];
+        snprintf(randfile, CF_BUFSIZE, "%s%crandseed",
+                 CFWORKDIR, FILE_SEPARATOR);
+
+        /* Only write out a seed if the file doesn't exist
+         * and we have enough entropy to do so. If RAND_write_File
+         * returns a bad value, delete the poor seed.
+         */
+        if (access(randfile, R_OK) && errno == ENOENT && RAND_write_file(randfile) != 1024)
+        {
+            Log(LOG_LEVEL_WARNING,
+                "Could not write randomness to '%s'", randfile);
+            unlink(randfile); /* do not reuse entropy */
+        }
+
+        chmod(randfile, 0600);
```

Comment:
Good catch this `chmod()`. This brought to my attention that the existing `chmod(644)` in KeepKeyPromises is bogus. Also I'd expect that OpenSSL is doing the right thing with permissions, if so, do you want to remove both chmod()s?